### DISCUSSION TITLE
Adding AUX Button

### DISCUSSION THREAD

**Discussion by sambondtm** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#707] [Q&A]
Hi All, apologies if there is an answer for this but i have tried searching already.

I want to connect a push to close button on my spare GPIO, in my case (D1 mini, "D1") , to operate my blower.

Ideally when i press that button i would like the blower to activate for 5 mins, if the button is pressed within the 5 min the blower turns off.

Of course i could use my phone to operate the blower (pump hidden under decking) but phones and water are not easy to use.

If someone can please guide me that would be amazing.

Thanks in advance.

**Comment by visualapproach** [comment]
Hi, you'd need to add some code for that. I prepared a small template in main.cpp loop IIRC. I'm choked now but I can give you some hint if you need later on.

**Reply by sambondtm** [reply]
Oh wow, thank you very much for your fast reply. 
The physical button could even do toggle the blower if that's easier ?

**Comment by chunkysteveo** [comment]
If you use Home Assistant - you can create a remote switch to do this. I made a 'Bubble button' using a zigbee sensor, and some automations in HA.

**Reply by sambondtm** [reply]
Thats a great idea thank you, but really wanted a standalone solution..

**Comment by sambondtm** [comment]
Hi there, do you think the following with the correct post for the toggle blower will work, or is there a better way ? sadly im away from home to try it:

#include <ESP8266HTTPClient.h>


const int inputPin = D1; // Pin connected to the input
const char* serverName = "http://192.168.0.5/post/example_command";

void setup() {
  pinMode(inputPin, INPUT);



void loop() {
  if (digitalRead(inputPin) == HIGH) {
    sendPostRequest();
    delay(1000); 
  }
}

void sendPostRequest() {
  if (WiFi.status() == WL_CONNECTED) { 
    HTTPClient http;

    http.begin(serverName); 
    http.addHeader("Content-Type", "application/x-www-form-urlencoded"); 

    int httpResponseCode = http.POST(""); 


    http.end(); 
  } 
}
